### ISSUE TITLE
The identifier parser should _not_ accept qualified reserved ops

### ISSUE THREAD

**Issue by harpocrates** [haskell/haddock#952]
This is really a GHC bug: https://ghc.haskell.org/trac/ghc/ticket/14109, but it causes Haddock to accept and render some impressive nonsense.

```haskell
module Temp where

-- | See 'Q.case', 'Q.of', 'Q.--', 'Q.as'
foo :: ()
foo = ()
```

results in `case`, `of`, `--`, and `as` attempting to be hyperlinked.

<img width="257" alt="screen shot 2018-10-23 at 8 39 59 am" src="https://user-images.githubusercontent.com/10766081/47372785-54768e00-d69f-11e8-91e9-ee38bea11d47.png">

**Comment by harpocrates** [comment]
Fixed in https://github.com/harpocrates/haddock/commit/6922aa95d01b1ee8d88174c083692c15cb408616. I'll figure where to slot this in (whether on `ghc-8.8`, `ghc-head`) after the dust settles.

**Comment by harpocrates** [comment]
Fixed in 9fcfa3715a1da307898a075668e3bc837595d406
